### PR TITLE
In release 1.8(not yet in mainnet), floor gas price should be checked…

### PR DESCRIPTION
**Pull request by arnabmitra**
… for all Tx's ( currently only being checked for Tx which have additional Msg Fee).

## Description

In release 1.8(not yet in mainnet), floor gas price should be checked for all Tx's ( currently only being checked for Tx which have additional Msg Fee)

closes: #684

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by fkneeland-figure — DISMISSED** [review]

File: internal/antewrapper/msg_fees_decorator.go [hunk 1]
```diff
@@ -108,6 +117,13 @@ func (afd MsgFeesDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool
 	return next(ctx, tx, simulate)
 }
 
+//   The chain-id check is exclusively for not breaking all existing sim tests which freak out when denom is anything other than stake.
+//   and some network tests won't work without a chain id being set(but they also setup everything with stake denom) so `simapp-unit-testing` chain id is skipped also.
+//   This only needs to work to pio-testnet and pio-mainnet, so this is safe.
+func chainIdsToIgnoreForFloorGasPriceCheck(ctx sdk.Context) bool {
```
Comment (channa-figure) [inline internal/antewrapper/msg_fees_decorator.go:123]: NIT: Can we rename this to something else?  At first look, I can't tell if it should return `true` or `false` by the name.  Maybe something like `shouldIgnoreFloorGasPriceCheck`?

**Review by channa-figure — CHANGES_REQUESTED** [review]

File: internal/antewrapper/msg_fees_decorator.go (hunk 1 above)
Comment (arnabmitra) [reply internal/antewrapper/msg_fees_decorator.go]: renamed

**Review by nullpointer0x00 — APPROVED** [review]

**Review by nullpointer0x00 — DISMISSED** [review]
LGTM!

**Review by fkneeland-figure — APPROVED** [review]
lgtm!

File: internal/antewrapper/msg_fees_decorator_test.go [hunk 2]
```diff
@@ -33,6 +33,28 @@ func (suite *AnteTestSuite) TestEnsureMempoolAndMsgFees() {
...
 }
 
+// checkTx true, high min gas price(high enough so that additional fee in same denom tips it over,
+//and this is what sets it apart from MempoolDecorator which has already been run)
+func (suite *AnteTestSuite) TestEnsureMempoolAndMsgFeesNoAdditionalFees() {
+	err, antehandler := setUpApp(suite, true, "nhash", 0)
+	tx, _ := createTestTx(suite, err, sdk.NewCoins(sdk.NewInt64Coin("nhash", 100000)))
+	suite.Require().NoError(err)
+
+	// Set gas price (1 atom)
```
Comment (iramiller) [inline internal/antewrapper/msg_fees_decorator_test.go:43]: this says `atom` but code says `nhash`

Comment (arnabmitra) [reply internal/antewrapper/msg_fees_decorator_test.go]: :doh: yes, i cleaned up the comments and added a new test.

**Comment by arnabmitra** [comment]
unfortunately i cleaned up a proto comment which was causing confusion too.

**Review by channa-figure — APPROVED** [review]

**Review by iramiller — APPROVED** [review]
Nice!

**Review by fkneeland-figure — APPROVED** [review]

**Pull request merged**